packages/adapter-node/src/index.js

```diff
@@ -6,10 +6,56 @@ export const path = env('SOCKET_PATH', false);
 export const host = env('HOST', '0.0.0.0');
 export const port = env('PORT', !path && '3000');
 
+const listen_pid = parseInt(env('LISTEN_PID', '-1'));
+const listen_fds = parseInt(env('LISTEN_FDS', '-1'));
+const timeout = parseInt(env('TIMEOUT', '-1'));
+const systemd_socket_fd = 3;
+
 const server = polka().use(handler);
 
-server.listen({ path, host, port }, () => {
-	console.log(`Listening on ${path ? path : host + ':' + port}`);
-});
+function close() {
+	// @ts-expect-error this was added in 18.2.0 but is not reflected in the types
+	server.server.closeIdleConnections();
+	server.server.close();
+}
+
+if (listen_pid === process.pid && listen_fds === 1) {
+	server.listen({ fd: systemd_socket_fd }, () => {
+		console.log('Listening on file descriptor 3');
```

Comment:
This is confusing - using a const of 3 but then having a magic number in the string. They should both by just literal '3's or both refer to the same constant imo.

Reply:
Yes you're right. I'm about to fix this.